model/project_ref.go

```diff
@@ -654,19 +663,60 @@ func (p *ProjectRef) AttachToNewRepo(u *user.DBUser) error {
 			return errors.Wrap(err, "adding project to new repo scope")
 		}
 	}
+
 	update := bson.M{
-		"$set": bson.M{
-			ProjectRefOwnerKey:     p.Owner,
-			ProjectRefRepoKey:      p.Repo,
-			ProjectRefRepoRefIdKey: p.RepoRefId,
-		},
+		ProjectRefOwnerKey:     p.Owner,
+		ProjectRefRepoKey:      p.Repo,
+		ProjectRefRepoRefIdKey: p.RepoRefId,
 	}
-	if err := db.UpdateId(ProjectRefCollection, p.Id, update); err != nil {
+	update = p.addGithubConflictsToUpdate(update)
+	err = db.UpdateId(ProjectRefCollection, p.Id, bson.M{
+		"$set": update,
+	})
+	if err != nil {
 		return errors.Wrap(err, "updating owner/repo in the DB")
 	}
 	return GetAndLogProjectModified(p.Id, u.Id, false, before)
 }
 
+// addGithubConflictsToUpdate turns off any settings that may introduce conflicts by
+// adding fields to the given update and returning them.
+func (p *ProjectRef) addGithubConflictsToUpdate(update bson.M) bson.M {
+	// If the project ref doesn't default to repo, will just return the original project.
+	mergedProject, err := GetProjectRefMergedWithRepo(*p)
+	if err != nil {
+		grip.Debug(message.WrapError(err, message.Fields{
```

Comment:
Chose not to error here since errors are unlikely, and erroring would require us to revert changes that could also error (similar to the case I described in the comments, I'm not sure it's worth all that). Open to discussion on it though.